### PR TITLE
Bump to MFEM 4.3

### LINKED ISSUE #532: Update MFEM to 4.3
MFEM 4.3 has been released: https://github.com/mfem/mfem/releases/tag/v4.3

We should upgrade and enable `hypre+cuda` to take advantage of the corresponding functionality in MFEM

### PR DESCRIPTION
**Pull request by joshessman-llnl**
Partially resolves #532 - I'll open a new issue for the hypre+cuda upgrade.

- [x] RZ hostconfigs
- [x] Docker hostconfigs

### CODE REVIEW

**Review by jamiebramwell — APPROVED** [review]
Thanks @joshessman-llnl !

File: scripts/spack/packages/mfem/package.py [hunk 1]
```diff
@@ -45,10 +45,15 @@ class Mfem(Package):
     # other version.
     version('develop', branch='master')
 
+    # SERAC EDIT BEGIN
+    version('4.3.0',
```
Comment (white238) [inline scripts/spack/packages/mfem/package.py:49]: Is this the exact 4.3 release? not a one of sha for us?

**Review by white238 — APPROVED** [review]

File: scripts/spack/packages/mfem/package.py (hunk 1 above)
Comment (joshessman-llnl) [reply scripts/spack/packages/mfem/package.py:49]: The SHA here is not a commit hash but the hash of the official release tarball - not sure if that's what you were asking

Comment (white238) [reply scripts/spack/packages/mfem/package.py:49]: oh yes its the checksum... carry on.

**Comment by jamiebramwell** [comment]
RZ host-configs incoming...

**Pull request merged**